### PR TITLE
Web Support

### PR DESCRIPTION
**Pull request by Vixeliz**
This is heavily wip no where near ready to be merged. Most examples don't work. Just want to have the draft open so I can ask for feedback when the time comes.

### CODE REVIEW

File: examples/bunnymark.rs [hunk 1]
```diff
@@ -163,6 +166,7 @@ impl event::EventHandler for GameState {
     ) -> GameResult {
         if button == input::mouse::MouseButton::Left && self.click_timer == 0 {
             for _ in 0..INITIAL_BUNNIES {
+                #[cfg(not(target_arch = "wasm32"))]
```
[the remark this answers is not in the document]
Comment (Vixeliz) [reply examples/bunnymark.rs:169]: Not enough instances can be drawn on web

File: src/coroutine.rs [hunk 2]
```diff
@@ -4,60 +4,154 @@
...
+/// Used to help with async loading of assets cleanly
+#[allow(missing_debug_implementations)]
+pub struct Loading<T, C = Context> {
+    pub(crate) coroutine: Coroutine<GameResult<T>, C>,
+    result: Option<T>,
+}
+
+impl<T, C> Loading<T, C>
+where
+    C: Has<Filesystem>,
```
[the remark this answers is not in the document]
Comment (Vixeliz) [reply src/coroutine.rs:88]: Hmmm ok, although genuinely wondering would loading ever *not* need the filesystem?

File: examples/bunnymark.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (nobbele) [reply examples/bunnymark.rs:169]: should you not just limit it to whatever the limit is then? native also has a limit

**Comment by Vixeliz** [comment]
Merging into a separate branch as I have less time to work on ggez now unexpectedly.

**Pull request merged**